File: hw4/Tests/CalculatorTests.cs
using hw4;
using Xunit;

namespace Tests
{
    public class CalculatorTests
    {
        [Theory]
        [InlineData(5,"+", 7, 12)]
        [InlineData(-8,"+", 3, -5)]
        public void Calculate_Plus(int val1, string operation, int val2, int expected)
        {
            var result = Calculator.calculate(val1, operation, val2);
            Assert.Equal(expected, result);
        }
        
        [Theory]
        [InlineData(49,"-", 53, -4)]
        [InlineData(-63,"-", -53, -10)]
        public void Calculate_Minus(int val1, string operation, int val2, int expected)
        {
            var result = Calculator.calculate(val1, operation, val2);
            Assert.Equal(expected, result);
        }
        [Theory]
        [InlineData(55,"*", 2, 110)]
        [InlineData(-8,"*", 0, 0)]
        public void Calculate_Multiply(int val1, string operation, int val2, int expected)
        {
            var result = Calculator.calculate(val1, operation, val2);
            Assert.Equal(expected, result);
        }
        [Theory]
        [InlineData(14,"/", 7, 2)]
        [InlineData(50,"/", 1, 50)]
        public void Calculate_Divide(int val1, string operation, int val2, int expected)
        {
            var result = Calculator.calculate(val1, operation, val2);
            Assert.Equal(expected, result);
        }
        
        [Theory]
        [InlineData(5,"9", 7, 0)]
        [InlineData(-8,"l", 3, 0)]
        public void Calculate_SomethingAnother(int val1, string operation, int val2, int expected)
        {
            var result = Calculator.calculate(val1, operation, val2);
            Assert.Equal(expected, result);
        }
    }
}

File: src/Homeworks/hw5/hw5.Tests/CalculatorTests.fs
module CalculatorTests
open hw5
open Parser
open Xunit
open Calculator

[<Theory>]
[<InlineData(16, "+", 15, 31)>]
[<InlineData(1, "-", 6, -5)>]
[<InlineData(5, "*", 3, 15)>]
[<InlineData(5, "+", 7, 12)>]
let ``Correct with int values`` (arg1, op, arg2, expected) =
    let result = Calculate arg1 op arg2
    Assert.Equal(Success expected, result)

[<Theory>]
[<InlineData(20.1, "+", 12.5, 32.6)>]
[<InlineData(2.5, "-", 1.6, 0.9)>]
[<InlineData(0.9, "*", 2, 1.8)>]
[<InlineData(8.23543, "/", 8.23543, 1)>]
let ``Correct with rational values`` (arg1, op, arg2, expected) =
    let result = Calculate arg1 op arg2
    Assert.Equal(Success expected, result)
    
[<Theory>]
[<InlineData(100, "/", 0)>]
let ``fail on division by zero`` (arg1, op, arg2) =
    let result = Calculate arg1 op arg2
    Assert.Equal(Failure "Divide by zero is not allowed.", result)